app/utils/IncomeSourcesUtils.scala

```diff
@@ -18,13 +18,19 @@ package utils
 
 import auth.MtdItUser
 import config.featureswitch.{FeatureSwitching, IncomeSources}
+import enums.IncomeSourceJourney.IncomeSourceType
+import enums.JourneyType.{Add, JourneyType}
+import models.incomeSourceDetails.AddIncomeSourceData
 import play.api.mvc.Result
 import play.api.mvc.Results.Redirect
+import services.SessionService
 import uk.gov.hmrc.auth.core.AffinityGroup.Agent
+import uk.gov.hmrc.http.HeaderCarrier
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 trait IncomeSourcesUtils extends FeatureSwitching {
```

Comment:
You may be able to avoid passing in sessionService and ec by doing this:

trait IncomeSourcesUtils extends FeatureSwitching {
  self =>

  implicit val ec: ExecutionContext

  val sessionService: SessionService

Reply:
Also there is a lot of unformatted code in this PR